add support for pmrun by dell

juliedavila opened the pull request:
##### ISSUE TYPE

Feature Pull Request
##### ANSIBLE VERSION

ansible 2.2.0 (pmrun c708c21e61) last updated 2016/06/16 17:49:57 (GMT -700)
  lib/ansible/modules/core: (detached HEAD 1d0f408897) last updated 2016/06/14 10:10:53 (GMT -700)
  lib/ansible/modules/extras: (detached HEAD 146c969c07) last updated 2016/06/14 10:10:53 (GMT -700)
  config file =
  configured module search path = Default w/o overrides
##### SUMMARY

Adds support for no-password pmrun priv escalation

Commit:
add support for pmrun by dell

bcoca commented on lib/ansible/playbook/play_context.py:
```diff
@@ -539,6 +539,17 @@ def detect_su_prompt(data):
                 becomecmd = '%s -u %s %s' % (exe, self.become_user, command)
 
+            elif self.become_method == 'pmrun':
+                #TODO: Add support for prompt
```
so 'Password:' is not a valid prompt? you should leave unset in that case

juliedavila replied:
it is a valid prompt

wtcross replied:
If this is the valid prompt, then isn't it going to work when pmrun prompts for a password? My understanding of this bit of code isn't very good, but it seems like you just set the `prompt` variable and then password input is handled for you.

@bcoca is this a correct assumption?

bcoca commented:
To use this it also requires whitelisting at the connection plugin level.

wtcross commented:
@bcoca @defionscode Do one of you have a way to test this code? I think that adding password support is a great idea, but we'll need to test that it's working before merging.

wtcross commented:
I'm happy to help out with this PR, btw.

Commit:
Whitelist pmrun at the connection plugin level.

This changes whitelists the pmrun privilege escalation
method at the connection plugin level, so that it can
be used.

Commit:
Merge pull request #1 from wtcross/pmrun

Whitelist pmrun at the connection plugin level.

wtcross commented:
The `pmrun` priv escalation method has been whitelisted at the connection plugin level.

juliedavila commented:
@wtcross looks like your pr to my fork caused some merge conflicts
